src/providers/tfc/runProvider.ts

```diff
@@ -4,15 +4,45 @@
  */
 
 import * as vscode from 'vscode';
+import { apiClient } from '../../terraformCloud';
+import { TerraformCloudAuthenticationProvider } from '../authenticationProvider';
+import axios from 'axios';
 import { WorkspaceTreeItem } from './workspaceProvider';
+import { RunAttributes } from '../../terraformCloud/run';
 
 export class RunTreeDataProvider implements vscode.TreeDataProvider<vscode.TreeItem>, vscode.Disposable {
   private readonly didChangeTreeData = new vscode.EventEmitter<void | vscode.TreeItem>();
   public readonly onDidChangeTreeData = this.didChangeTreeData.event;
+  private activeWorkspaceId: string | undefined;
 
-  constructor(private ctx: vscode.ExtensionContext) {}
+  constructor(private ctx: vscode.ExtensionContext, private workspaceView: vscode.TreeView<WorkspaceTreeItem>) {
+    this.ctx.subscriptions.push(vscode.commands.registerCommand('terraform.cloud.runs.refresh', () => this.refresh()));
+
+    // TODO: get from settings or somewhere global
+    const baseUrl = 'https://app.staging.terraform.io/app';
+
+    this.ctx.subscriptions.push(
+      vscode.commands.registerCommand('terraform.cloud.run.view', (run: RunTreeItem) => {
+        const orgName = this.ctx.workspaceState.get('terraform.cloud.organization', '');
+        if (orgName === '') {
+          return;
+        }
+
+        const runURL = `${baseUrl}/${orgName}/workspaces/${run.workspaceName}/runs/${run.id}`;
```

Comment:
We don't need as much info here as you're collecting. Each run has a link attribute with a link to itself https://developer.hashicorp.com/terraform/cloud-docs/api-docs/run#list-runs-in-a-workspace

Reply:
I also don't like the idea of building out the URL here, but AFAICT that self link is unfortunately just a link to the API, not the "browser" version.

```json
"links": {
    "self": "/api/v2/runs/run-bWSq4YeYpfrW4mx7"
}
```